lib/Dialect/SV/Transforms/HWCleanup.cpp

```diff
@@ -245,6 +342,97 @@ void HWCleanupPass::runOnProceduralRegion(Region &region) {
   }
 }
 
-std::unique_ptr<Pass> circt::sv::createHWCleanupPass() {
-  return std::make_unique<HWCleanupPass>();
+// This function merge two if operations when they have a common condition.
+// For example,
+// prevIfOp:
+//   if a1 & a2 &... & an & c1 & c2 .. & cn {e1}
```

Comment:
It seems that CSE should first have split the c terms into their own expression, then the logic here would be whether two ifs share a common term which is non-trivial.  This would have the benefit of catching
a & ... & (c | d | e) with b & ... & (c | d | e)

Reply:
I think it is difficult to rely on CSE to split only c terms since it is requires CSE to split variadic operands. For example, the following operations will not be splitted easily. 
 
```
%a = comb.and %0, %1, %2 : i1
%b = comb.and %0, %1 : i1
%c = comb.and %0, %2 : i1
%d = comb.and %1, %2 : i1
```

This hoisting transformation is heavily depending on the restriction that we merge successive if-ops.